examples/jlang-parser/src/main.rs

```diff
@@ -0,0 +1,261 @@
+    MonadicOp { verb: MonadicVerb, expr: Box<AstNode> },
+    DyadicOp { verb: DyadicVerb, lhs: Box<AstNode>, rhs: Box<AstNode>},
+    Terms(Vec<AstNode>),
+    Reduce { verb: DyadicVerb, expr: Box<AstNode> },
+    IsGlobal{ident: String, expr: Box<AstNode>},
+    Ident(String),
+    Str(CString),
+}
+
+impl std::fmt::Display for AstNode {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        let _ = write!(f, "{:?}", self);
+        Ok(())
+    }
+}
+
+pub fn parse(source: &str) -> Result<Vec<AstNode>, Error<Rule>> {
+    let mut ast = vec![];
+
+    let pairs = JParser::parse(Rule::program, source)?;
+    for pair in pairs {
+        match pair.as_rule() {
+            Rule::expr => {
+                ast.push(Print(Box::new(build_ast_from_expr(pair))));
+            }
+            _ => {}
+        }
+    }
+
+    Ok(ast)
+}
+
+fn build_ast_from_expr(pair: pest::iterators::Pair<Rule>) -> AstNode {
+
+    match pair.as_rule() {
+        Rule::expr => build_ast_from_expr(pair.into_inner().next().unwrap()),
+        Rule::monadicExpr => {
+            let mut pair = pair.into_inner();
+            let action = pair.next().unwrap();
+            let expr = pair.next().unwrap();
+            let expr = build_ast_from_expr(expr);
+            parse_monadic_action(action, expr)
+        },
+        Rule::dyadicExpr => {
+            let mut pair = pair.into_inner();
+            let lhspair = pair.next().unwrap();
+            let lhs = build_ast_from_expr(lhspair);
+            let action = pair.next().unwrap();
+            let rhspair = pair.next().unwrap();
+            let rhs = build_ast_from_expr(rhspair);
+            parse_dyadic_action(action, lhs, rhs)
+        },
+        Rule::terms => {
+            let terms : Vec<AstNode>= pair.into_inner()
+                .map(build_ast_from_term)
+                .collect();
+            // If there's just a single term, return it without
+            // wrapping it in a Terms node.
+            match terms.len() {
+                1 => terms.get(0).unwrap().clone(),
+                _ => Terms(terms),
+            }
+        },
+        Rule::assgmtExpr => {
+            let mut pair = pair.into_inner();
+            let ident = pair.next().unwrap();
+            let expr = pair.next().unwrap();
+            let expr = build_ast_from_expr(expr);
+            AstNode::IsGlobal { ident : String::from(ident.as_str()),
+                expr : Box::new(expr) }
+        },
+        Rule::string => {
+            let str = &pair.as_str();
+            // Strip leading and ending quotes.
+            let str = &str[1..str.len() - 1];
+            // Escaped string quotes become single quotes here.
+            let str = str.replace("''", "'");
+            AstNode::Str(CString::new(&str[..]).unwrap())
+        }
+        unknown_expr => panic!("Unexpected expression: {:?}", unknown_expr),
+    }
+}
+
+fn parse_dyadic_action(pair : pest::iterators::Pair<Rule>,
```

Comment:
Could you please use the default rustfmt formatting here? pest has a [configuration](https://github.com/pest-parser/pest/blob/master/rustfmt.toml) that will probably work fine, even though it's basically rustfmt default.